### DISCUSSION TITLE
Way to createSignedUrls with image transformation

### DISCUSSION THREAD

**Discussion by devidevio** [supabase/supabase#12607] [Feature Requests]
I've checked every part of documentation and code in the storage-api regarding the `createSignedUrls` function. It seems impossible to sign multiple URLs with image transformation.

Related documentation:
* https://supabase.com/docs/reference/javascript/storage-from-createsignedurls
* https://supabase.com/docs/reference/self-hosting-storage/generate-a-presigned-url-to-retrieve-an-object
* https://supabase.com/docs/guides/storage/image-transformations

The code in the storage-api seems to be the following:
* https://github.com/supabase/storage-api/blob/eced90e45b481021332c7f63d61b068f54fab963/src/storage/object.ts
* https://github.com/supabase/storage-api/blob/eced90e45b481021332c7f63d61b068f54fab963/src/storage/object.ts#L411
* https://github.com/supabase/storage-api/blob/59f99262b7c31740dd7b74cb6858648135c9168e/src/http/routes/object/getSignedURL.ts#L80

Why does the `createSIgnedURLs` function does not accept the image transformations when signing a URL?

The reason why I'm looking for this feature is: Performance. Instead of signing f. e. 100 URLs separately (100 requests), I'd like to sign 100 URLs in one request with image transformation.

If I'm missing something at this point here, please correct me.

**Comment by edeesis** [comment]
Found this question wanting the same thing (3+ years later..)

[Looks like createSignedUrls](https://github.com/supabase/storage/blob/eced90e45b481021332c7f63d61b068f54fab963/src/storage/object.ts#L450) does nothing more than a Promise.all over the paths array with `n` network requests, so there's no performance benefit for it, just a slightly nicer wrapper.

Looks like there's no downside to a Promise.all with the transform for our use case.
